ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFDateFormat.java

```diff
@@ -106,25 +102,18 @@ public Object evaluate(DeferredObject[] arguments) throws HiveException {
       return null;
     }
 
-    ZoneId id = (SessionState.get() == null) ? new HiveConf().getLocalTimeZone() : SessionState.get().getConf()
-        .getLocalTimeZone();
     // the function should support both short date and full timestamp format
     // time part of the timestamp should not be skipped
     Timestamp ts = getTimestampValue(arguments, 0, tsConverters);
+
     if (ts == null) {
-      Date d = getDateValue(arguments, 0, dtInputTypes, dtConverters);
-      if (d == null) {
-        return null;
-      }
-      ts = Timestamp.ofEpochMilli(d.toEpochMilli(id), id);
+      return null;
```

Comment:
Why DateConverter is removed? Isn't it needed to convert input like "2021-06-24" or does it work with TimestampCoverter?

Reply:
It works with Timestamp converter